File: engine-server/app/models/product.rb
class Product < ApplicationRecord
	has_many :matches
	has_many :gemini_matches
	has_many :kraken_matches
	validates_uniqueness_of   :product_name
	def self.save_products(product_array)
		current_products = []
		product_array.each do |product|
			current_products << Product.find_or_create_by(product_name: product["id"]) do |p|
				p.base_currency = product["base_currency"]
				p.quote_currency = product["quote_currency"]
				p.base_min_size = product["base_min_size"]
				p.base_max_size = product["base_max_size"]
				p.quote_increment = product["quote_increment"]
				p.display_name = product["display_name"]
				p.margin_enabled = product["margin_enabled"]
				p.status = product["status"]
				p.status_message = product["status_message"]
				p.gdax = true
			end
		end
		return current_products
	end
end

File: engine-client/src/app/product.ts
import { CommonModule } from '@angular/common';

export class Product {
    id: string;
    product_name: string
    base_currency: string;
    quote_currency: string;
    base_min_size: number;
    base_max_size: number;
    quote_increment: number;
    display_name: string;
    status: string;
    margin_enabled: boolean;
    status_message: string;
    // This number will change based on the quote currency
    price: any;
    k_price: any;
    g_price: any;
    c_price: any;
    time: any;
    c_time: any;
    k_time: any;
    g_time: any;
    constructor(id:string){
        this.id = id;
    }

    
}
